src/api/ml_modules/extractive_summarizer_model.py

```diff
@@ -8,46 +8,25 @@
 def process_text(text, nlp):
     doc = nlp(text)
     word_frequencies = {}
-
-    for word in doc:
-        if word.text.lower() not in list(STOP_WORDS):
-            if word.text.lower() not in punctuation:
-                if word.text not in word_frequencies.keys():
-                    word_frequencies[word.text] = 1
-                else:
-                    word_frequencies[word.text] += 1
+    word_frequencies = {word.text: word_frequencies.get(word.text, 0) + 1 for word in doc if word.text.lower()
+                        not in list(STOP_WORDS) and word.text.lower() not in punctuation}
```

Comment:
podziel na mniejsze funkcje. np to co po if-ie można dodac do jakiejś funkcji ktora konkretnie będzie mówic co sprawdza